purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -449,8 +440,21 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
             appUserID,
             { entitlements ->
                 getSkuDetails(entitlements, { detailsByID ->
-                    cachedEntitlements = entitlements
-                    populateSkuDetailsAndCallCompletion(detailsByID, entitlements, completion)
+                    entitlements.values.flatMap { it.offerings.values }.let { offerings ->
+                        val missingProducts= populateSkuDetails(offerings, detailsByID)
+                        if (missingProducts.isNotEmpty()) {
+                            log("Could not find SkuDetails for ${missingProducts.joinToString(", ")}")
+                            log("Ensure your products are correctly configured in Play Store Developer Console")
+                        }
+                        if (offerings.mapNotNull { it.skuDetails }.isNotEmpty()) {
+                            cachedEntitlements = entitlements
+                        } else {
+                            cachedEntitlements = null
+                        }
```

Comment:
if it can't find any skudetails, it will retry next time, that's why this change. it was saving as cachedEntitlements sku details that could be null